pkg/debug/networkchaos/networkchaos.go

```diff
@@ -0,0 +1,165 @@
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, ioutil.Discard, ioutil.Discard))
+	pid, err := cm.GetPidFromPod(ctx, pod, daemon)
+	if err != nil {
+		return err
+	}
+	nsenterPath := "-n/proc/" + strconv.Itoa(pid) + "/ns/net"
+
+	// print out debug info
+	cmd := fmt.Sprintf("/usr/bin/nsenter %s -- ipset list", nsenterPath)
+	out, err := cm.Exec(daemonName, daemonNamespace, cmd, c.K8sClient)
+	if err != nil {
+		return fmt.Errorf("run command '%s' failed with: %s", cmd, err.Error())
+	}
+	cm.Print("1. [ipset list]", 1, cm.ColorCyan)
+	cm.Print(string(out), 1, "")
+
+	cmd = fmt.Sprintf("/usr/bin/nsenter %s -- tc qdisc list", nsenterPath)
+	out, err = cm.Exec(daemonName, daemonNamespace, cmd, c.K8sClient)
+	if err != nil {
+		return fmt.Errorf("run command '%s' failed with: %s", cmd, err.Error())
+	}
+	cm.Print("2. [tc qdisc list]", 1, cm.ColorCyan)
+	cm.Print(string(out), 1, "")
+
+	action := chaos.Spec.Action
+	var netemExpect string
+	switch action {
+	case "delay":
+		latency := chaos.Spec.Delay.Latency
+		jitter := chaos.Spec.Delay.Jitter
+		correlation := chaos.Spec.Delay.Correlation
+		netemExpect = fmt.Sprintf("%v %v %v %v%%", action, latency, jitter, correlation)
+	default:
+		return fmt.Errorf("chaos not supported")
+	}
+
+	netemCurrent := regexp.MustCompile("(?:limit 1000)(.*)").FindStringSubmatch(string(out))
+	if len(netemCurrent) == 0 {
+		return fmt.Errorf("No NetworkChaos is applied")
+	}
+	for i, netem := range strings.Fields(netemCurrent[1]) {
+		itemCurrent := netem
+		itemExpect := strings.Fields(netemExpect)[i]
+		if itemCurrent != itemExpect {
+			r := regexp.MustCompile("([0-9]*[.])?[0-9]+")
+			numCurrent, err := strconv.ParseFloat(r.FindString(itemCurrent), 64)
+			if err != nil {
+				return fmt.Errorf("parse itemCurrent failed: %s", err.Error())
+			}
+			numExpect, err := strconv.ParseFloat(r.FindString(itemExpect), 64)
+			if err != nil {
+				return fmt.Errorf("parse itemExpect failed: %s", err.Error())
+			}
+			if numCurrent == numExpect {
+				continue
+			}
+			alpCurrent := regexp.MustCompile("[[:alpha:]]+").FindString(itemCurrent)
+			alpExpect := regexp.MustCompile("[[:alpha:]]+").FindString(itemExpect)
+			if alpCurrent == alpExpect {
+				continue
+			}
+			errInfo := fmt.Sprintf("NetworkChaos didn't execute as expected, expect: %s, got: %s", netemExpect, netemCurrent)
+			cm.Print(errInfo, 1, cm.ColorRed)
+			return nil
+		}
+	}
+	cm.Print("NetworkChaos execute as expected", 1, cm.ColorGreen)
+
+	cmd = fmt.Sprintf("/usr/bin/nsenter %s -- iptables --list", nsenterPath)
+	out, err = cm.Exec(daemonName, daemonNamespace, cmd, c.K8sClient)
+	if err != nil {
+		return fmt.Errorf("cmd.Run() failed with: %s", err.Error())
+	}
+	cm.Print("3. [iptables list]", 1, cm.ColorCyan)
+	cm.Print(string(out), 1, "")
+
+	cmd = fmt.Sprintf("/usr/bin/nsenter %s -- iptables --list", nsenterPath)
+	out, err = cm.Exec(daemonName, daemonNamespace, cmd, c.K8sClient)
```

Comment:
Seems the output of `iptables` hasn't be used.

Reply:
Ah my bad fixed thx!